File: app/src/test/java/com/commit451/gitlab/TestApp.java
package com.commit451.gitlab;

import org.robolectric.TestLifecycleApplication;

import java.lang.reflect.Method;

/**
 * Test version of our Application class, used by Robolectric
 */
public class TestApp extends App implements TestLifecycleApplication {

    @Override
    public void beforeTest(Method method) {

    }

    @Override
    public void prepareTest(Object test) {

    }

    @Override
    public void afterTest(Method method) {

    }

    @Override
    protected void setupCrashReporting() {
        //Intentionally left blank
    }

    @Override
    protected void setupLeakCanary() {
        //Intentionally left blank
    }

    @Override
    protected void setupMultidex() {
        //Intentionally left blank
    }
}

File: app/src/test/java/com/commit451/gitlab/TestApp.kt
package com.commit451.gitlab

import org.robolectric.TestLifecycleApplication

import java.lang.reflect.Method

/**
 * Test version of our Application class, used by Robolectric
 */
class TestApp : App(), TestLifecycleApplication {

    override fun beforeTest(method: Method) {

    }

    override fun prepareTest(test: Any) {

    }

    override fun afterTest(method: Method) {

    }

    override fun setupCrashReporting() {
        //Intentionally left blank
    }

    override fun setupLeakCanary() {
        //Intentionally left blank
    }

    override fun setupMultidex() {
        //Intentionally left blank
    }

    override fun setupThreeTen() {
        //Intentionally left blank
    }
}
